mesonbuild/backend/backends.py

```diff
@@ -1278,6 +1278,7 @@ def create_test_serialisation(self, tests: T.List['Test']) -> T.List[TestSeriali
                 extra_bdeps: T.List[T.Union[build.BuildTarget, build.CustomTarget, build.CustomTargetIndex]] = []
                 if isinstance(exe, build.CustomTarget):
                     extra_bdeps = list(exe.get_transitive_build_target_deps())
+                extra_bdeps.extend(t.depends)
```

Comment:
You could probably replace the next loop by something like:
`extra_bdeps.extend(a for a in t.cmd_args if isinstance(a, build.BuildTarget)`

That would allow to call `determine_windows_extra_paths` only once.

Reply:
Very nice!